tests/Silverback.Integration.Tests.E2E/Kafka/KafkaBrokerEventsTests.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Silverback.Messaging.Configuration;
+using Silverback.Messaging.Messages;
+using Silverback.Messaging.Publishing;
+using Silverback.Testing;
+using Silverback.Tests.Integration.E2E.TestHost;
+using Silverback.Tests.Integration.E2E.TestTypes.Messages;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Silverback.Tests.Integration.E2E.Kafka
+{
+    public class KafkaBrokerEventsTests : KafkaTestFixture
+    {
+        public KafkaBrokerEventsTests(ITestOutputHelper testOutputHelper)
+            : base(testOutputHelper)
+        {
+        }
+
+        [Fact]
+        public async Task BrokerEventsRegistered_EndpointsConfiguredHandlerCalled()
+        {
+            Host.ConfigureServices(
+                    services => services
+                        .AddLogging()
+                        .AddSilverback()
+                        .UseModel()
+                        .WithConnectionToMessageBroker(
+                            options => options
+                                .AddMockedKafka()
+                                .AddSingletonBrokerEventsHandler<SimpleBrokerEventsHandler>())
```

Comment:
Seeing the final result I just realized that having the method in here prevents it to be configured at feature level (I think). Is this an issue in your opinion?

Or just to be safe we could move it directly at ISilverbackBuilder level (just change the extended type of the extension methods). So that if can be called at the same level of `Add*Subscriber` and `Add*EndpointsConfigurator`.

Reply:
That's a good input. I will move it to `ISilverbackBuilder` level.